src/pdf_parser/pdf_utils/parsing_utils.py

```diff
@@ -606,34 +570,191 @@ def _infer_column_groups(self, threshold: float = 0.95):
+                        continue
+                    else:
+                        permissive_block_vline = LineString(
+                            [
+                                (block_2.coordinates[1], 0),
+                                (block_2.coordinates[3], 0),
+                            ]
+                        )
+                        line_intersection = permissive_block_vline.intersection(
+                            restrictive_block_vline
+                        )
+                        if line_intersection.is_empty:
+                            continue
+                        else:
+                            unaccounted_vline = permissive_block_vline.difference(
+                                line_intersection
+                            )
+                            if (
+                                unaccounted_vline.bounds
+                                == permissive_block_vline.bounds
+                            ) or (unaccounted_vline.is_empty):
+                                continue
+                            else:
+                                unaccounted_vlines.append(unaccounted_vline)
+                if len(unaccounted_vlines) == 0:
+                    continue
+                else:
+                    unaccounted_vline = unary_union(unaccounted_vlines)
+                    y1, y2 = unaccounted_vline.bounds[0], unaccounted_vline.bounds[2]
+                    x1, x2 = block_1.coordinates[0], block_1.coordinates[2]
+                    unaccounted_block_shape = lp.Rectangle(x1, y1, x2, y2)
+                    # TODO: include metadata such as a score in the text block?
+                    unaccounted_text_block = lp.TextBlock(
+                        unaccounted_block_shape, type="Ambiguous"
+                    )
+                    new_text_blocks.append(unaccounted_text_block)
+        return new_text_blocks
+
+    # TODO: Instead of this, we can use Google OCR to extract all text from a page and index everything not indexed.
+    #  TODO: Noticed a bug that needs fixing with high priority. This function fails because it doesn't account for
+    #   columns - if we're applying vertically we must do it separately for each column. This is nontrivial as it
+    #   will change the execution order.
+    def postprocess(
+        self,
+        inference_method: str = "gaps",
+    ) -> lp.Layout:
+        """Create text blocks by grabbing blocks with low confidence scores and subtracting intersections
+         with the current layout to create new text blocks.
+
+        This has utility beyond the _combine_layouts method because it extracts text even in cases where
+        there are huge text blocks with high emounts of explained area so they aren't caught by the
+        _combine_layouts method.
+
+        Args:
+            inference_method: The method to use for inferring the missing blocks. Options are "gaps" and "threshold". See docstrings for details.
```

Comment:
might be worth putting a short note somewhere in the docstring with what you were just saying about why gaps is preferred to threshold to explain the default